langkit/support/langkit_support-diagnostics-output.ads

```diff
@@ -0,0 +1,88 @@
+--                                 Langkit                                  --
+--                                                                          --
+--                       Copyright (C) 2020, AdaCore                        --
+--                                                                          --
+-- Langkit is free software; you can redistribute it and/or modify it under --
+-- terms of the  GNU General Public License  as published by the Free Soft- --
+-- ware Foundation;  either version 3,  or (at your option)  any later ver- --
+-- sion.   This software  is distributed in the hope that it will be useful --
+-- but WITHOUT ANY WARRANTY;  without even the implied warranty of MERCHAN- --
+-- TABILITY  or  FITNESS  FOR A PARTICULAR PURPOSE.                         --
+--                                                                          --
+-- As a special  exception  under  Section 7  of  GPL  version 3,  you are  --
+-- granted additional  permissions described in the  GCC  Runtime  Library  --
+-- Exception, version 3.1, as published by the Free Software Foundation.    --
+--                                                                          --
+-- You should have received a copy of the GNU General Public License and a  --
+-- copy of the GCC Runtime Library Exception along with this program;  see  --
+-- the files COPYING3 and COPYING.RUNTIME respectively.  If not, see        --
+-- <http://www.gnu.org/licenses/>.                                          --
+------------------------------------------------------------------------------
+
+--  This package supports outputting ``Langkit_Support.Diagnostics.Diagnostic``
+--  for a source buffer, in a pretty colorized format akin to GCC's verbose
+--  diagnostic format.
+--
+--  .. note::
+--
+--      For the moment, this package has a few limitations, namely:
+--
+--      - Only source buffers with LF line endings are handled.
+--      - Only error messages spanning one line are gracefully handled.
+--        eventually we want to have something more powerful that has a syntax
+--        for error spanning multiple lines.
+--
+--  .. todo::
+--
+--      For the moment this is only used in the ``lkt_toolbox`` executable.
+--      Eventually, this should be exposed to generated library users in some
+--      fashion.
+
+package Langkit_Support.Diagnostics.Output is
+   type Text_Buffer is private;
+   --  This type is used as a wrapper around text, where we'll keep line
+   --  information to avoid re-processing it everytime we need it.
+
+   function Create (Text : Text_Cst_Access) return Text_Buffer;
+   --  Create a text buffer from a pointer to text
+
+   function Get_Line
+     (Self        : in out Text_Buffer;
+      Line_Number : Positive) return Text_Type;
+   --  Get line ``Line_Number``
+
+   procedure Print_Diagnostic
+     (Self : Diagnostic; Buffer : in out Text_Buffer; Path : String);
+   --
```

Comment:
I suggest to remove this empty line (for consistency with the rest of the codebase)